File: scripts/do_something.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import json

def do_something(input_file, output_file):
    """
    Do something
    """
    
    blah = {}

    with open(output_file, 'w') as fout:
        json.dump(blah, fout)


if __name__ == '__main__':
    # redirect all output to the log
    sys.stderr = open(snakemake.log[0], 'w')
    
    do_something(
		input_file=snakemake.input[0],
        output_file=snakemake.output[0]
    )

File: scripts/do_something.R
#!/usr/bin/env Rscript

suppressMessages(library(rjson, quietly=T, warn.conflicts=F))

# Do something
#
do_something <- function(input_file, output_file) {
	blah = c()

    cat(toJSON(blah, indent = 2), file=output_file)
}


if (sys.nframe() == 0) {
	# redirect all output to the log
	log <- file(snakemake@log[[1]], open="wt")
	sink(log)

	do_something(
		input_file=snakemake@input[[1]],
		output_file=snakemake@output[[1]]
	)
}
